### PR TITLE
Add --skip-magic-trailing-comma

### PR DESCRIPTION
**Pull request by hauntsaninja**
Lukasz indicated he'd accept such a PR in https://github.com/psf/black/issues/1288#issuecomment-593873891

This is a pretty straightforward combination of things from before and after #1612 and #826

Feedback I'm looking for:
1) What the best way to get the MAGIC_TRAILING_COMMA option to the places it's needed is.
Obviously the global variable proof of concept is unmergeable, but I'm not sure what the right solution is. Passing it around seems kind of untenable because `Leaf.append` is used in a million places / `should_split_body_explode` has a daunting reverse call graph too (You could make store an attribute on `Leaf` to make things a little easier, but that's pretty ugly as well). Maybe a context manager to set a global current `Mode`?
2) How best to structure tests for this 
Black doesn't have many options, so there isn't a clear pattern for how to test these. I went with reusing expression.py, but I could do something more akin to `numeric_literals_skip_underscores`. Let me know!

### CODE REVIEW

**Comment by ambv** [comment]
You're right that using global variables won't be acceptable for this. Hiding the global variable by running a context manager is not fundamentally changing this either.

Having an attribute on Leaf is what I would look at.

**Comment by hauntsaninja** [comment]
Thanks! You'll notice from the diff that I meant to say `Line`, not `Leaf` — hope it's still acceptable :-) I also updated blackd and the relevant docs that I could find.

**Comment by graingert** [comment]
@hauntsaninja thanks for following up on this, I'd completely missed the concept ack on the other issue

**Comment by hauntsaninja** [comment]
@ambv if you have time to take a look, much obliged!

**Comment by hauntsaninja** [comment]
Just bumping this!

File: docs/the_black_code_style.md [hunk 1]
```diff
@@ -438,6 +438,9 @@ into one item per line.
 How do you make it stop? Just delete that trailing comma and _Black_ will collapse your
 collection into one line if it fits.
 
+If you must, you can recover the behaviour of early versions of Black with the option
```
Comment (JelleZijlstra) [inline docs/the_black_code_style.md:441]: we might need to be consistent here at some point:
```
(py38) mbpt-jelle:black jelle$ git grep behavior | wc -l
       7
(py38) mbpt-jelle:black jelle$ git grep behaviour | wc -l
       7
```

**Review by JelleZijlstra — APPROVED** [review]

**Comment by uriva** [comment]
any idea when will this be released?

**Comment by ichard26** [comment]
@uriva unfortunately I can't give you a timeframe, but I do hope it will happen soon! Apologies for the lack of any clear messaging.

**Comment by uriva** [comment]
@ichard26 what is this blocked on? or is releasing itself a difficult process? (not demanding anything, just curious)

**Comment by ichard26** [comment]
@uriva I honestly don't know 100% why a new release is blocked, but the general lack of time and specifically the lack of availability for the ones holding the power to create a release is probably why. Maybe @cooperlees can explain in more detail.

**Comment by uriva** [comment]
who has this power?

**Comment by graingert** [comment]
@uriva you can see the list here https://pypi.org/project/black/#data

**Comment by indigoviolet** [comment]
tangential comment: I found this option `--skip-trailing-magic-comma` in the github documentation, but then didn't find it in the most recent release and it was mildly confusing. It might be useful to either have per-feature metadata like `[Added in 20.0b2dev]` etc or a `develop` branch.

**Comment by ichard26** [comment]
@indigoviolet you're not alone! I find it confusing too. I am planning to just move the documentation to ReadTheDocs where the stable version will be provided by default to avoid this mess in the first place (it's part of my effort to revamp the docs). Thanks for the feedback though!

**Comment by hauntsaninja** [comment]
Opened https://github.com/psf/black/pull/2064 to add this to CHANGES.md

Separately, is there anything blocking a release that I could help with?

**Pull request merged**